linux/hci/gap.go

```diff
@@ -206,31 +206,39 @@ func (h *HCI) Dial(ctx context.Context, a ble.Addr) (ble.Client, error) {
 		return nil, err
 	}
 	var tmo <-chan time.Time
-	if h.dialerTmo != time.Duration(0) {
+	if deadline, ok := ctx.Deadline(); ok {
+		tmo = time.After(time.Until(deadline))
```

Comment:
I'm not sure if this is necessary since if the context has a deadline, then it is already effectively a timeout. If the goal is to override the dialer timeout, then I think the select on `<-ctx.Done()` could be removed since selecting `<-tmo` is the same thing. Otherwise, I think the changes look great.